op-node/rollup/derive/system_config.go

```diff
@@ -36,18 +36,38 @@ var (
 // UpdateSystemConfigWithL1Receipts filters all L1 receipts to find config updates and applies the config updates to the given sysCfg
 func UpdateSystemConfigWithL1Receipts(sysCfg *eth.SystemConfig, receipts []*types.Receipt, cfg *rollup.Config, l1Time uint64) error {
 	var result error
+
+	// copy sysConfig to an update accumulator
+	// to allow for pre-parsing of all updates prior to application
+	updated := eth.SystemConfig{
+		BatcherAddr:          sysCfg.BatcherAddr,
+		Overhead:             sysCfg.Overhead,
+		Scalar:               sysCfg.Scalar,
+		GasLimit:             sysCfg.GasLimit,
+		EIP1559Params:        sysCfg.EIP1559Params,
+		OperatorFeeParams:    sysCfg.OperatorFeeParams,
+		MinBaseFee:           sysCfg.MinBaseFee,
+		DAFootprintGasScalar: sysCfg.DAFootprintGasScalar,
+	}
 	for i, rec := range receipts {
 		if rec.Status != types.ReceiptStatusSuccessful {
 			continue
 		}
 		for j, log := range rec.Logs {
 			if log.Address == cfg.L1SystemConfigAddress && len(log.Topics) > 0 && log.Topics[0] == ConfigUpdateEventABIHash {
-				if err := ProcessSystemConfigUpdateLogEvent(sysCfg, log, cfg, l1Time); err != nil {
+				if err := ProcessSystemConfigUpdateLogEvent(&updated, log, cfg, l1Time); err != nil {
+					if errors.Is(err, ErrParsingSystemConfig) {
+						// if the config can't be parsed, ignore the entire update
+						// an unparsable update should be treated as unusable and thus ignored, there are no other recovery mechanisms
+						return nil
+					}
```

Comment:
should we put this `early return` after the error is appended to result on line 64?

Reply:
No, because we don't want to return any errors at all if the config can't be updated, as its now being updated atomically.
- If all receipts parse and apply and there are no errors, great.
- If any of the updates create an error (besides parsing), then the error gets accumulated and returned on line 64 like you say
- If any of the Receipts can't be parsed, *no update* should be done, and so no other errors could exist

Suppose you had 3 receipts you were using to build updates: R1, R2, R3. If R1 would cause an error, but R2 is malformed, then R1's error is irrelevant, we will ignore the malformed update altogether, and not even apply R1.